### PR TITLE
ci: skip unnecessary tests

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

# Description

Skip unnecessary tests like https://github.com/dapr/cli/pull/749, speed CI up and decrease pressure.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: .github/workflows/installdaprwin.yml [hunk 1]
```diff
@@ -20,9 +20,21 @@ on:
...
+  pre_job:
+    name: Skip Duplicate Actions
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v3.4.0
+        with:
+          paths_ignore: '["**.md", ".codecov.yaml", ".github/**/**.yml"]'
```
Comment (wcs1only) [inline .github/workflows/installdaprwin.yml:33]: Same as above.

File: .github/workflows/dapr_cli.yml [hunk 2]
```diff
@@ -17,8 +17,19 @@ on:
...
+  pre_job:
+    name: Skip Duplicate Actions
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v3.4.0
+        with:
+          paths_ignore: '["**.md", ".codecov.yaml", ".github/**/**.yml", "CODEOWNERS"]'
```
Comment (wcs1only) [inline .github/workflows/dapr_cli.yml:29]: Why `.github/**/**.yml`? We absolutely want to run the tests when we make changes to the github actions workflows. How else will we iterate on them?

File: .github/workflows/self_hosted_e2e.yml [hunk 3]
```diff
@@ -18,8 +18,20 @@ on:
...
+  pre_job:
+    name: Skip Duplicate Actions
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v3.4.0
+        with:
+          paths_ignore: '["**.md", ".codecov.yaml", ".github/**/**.yml", "CODEOWNERS"]'
```
Comment (wcs1only) [inline .github/workflows/self_hosted_e2e.yml:30]: Same as above.

File: .github/workflows/upgrade_e2e.yml [hunk 4]
```diff
@@ -23,6 +23,17 @@ defaults:
...
+  pre_job:
+    name: Skip Duplicate Actions
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v3.4.0
+        with:
+          paths_ignore: '["**.md", ".codecov.yaml", ".github/**/**.yml", "CODEOWNERS"]'
```
Comment (wcs1only) [inline .github/workflows/upgrade_e2e.yml:35]: Same as above

File: .github/workflows/dapr_cli.yml [hunk 5]
```diff
@@ -54,53 +65,62 @@ jobs:
           go-version: ${{ env.GOVER }}
       - name: Check out code into the Go module directory
         uses: actions/checkout@v2
+      - name: Check if need skip
+        id: skip_check
+        uses: fkirc/skip-duplicate-actions@v3.4.0
+        with:
+          paths_ignore: '["**.md", ".codecov.yaml", ".github/**/**.yml", "CODEOWNERS"]'
```
Comment (wcs1only) [inline .github/workflows/dapr_cli.yml:72]: Same as above

File: .github/workflows/installdaprwin.yml (hunk 1 above)
Comment (daixiang0) [reply .github/workflows/installdaprwin.yml:33]: Not all, some like stale bot would work before merge and tests for those changes make no sense. 

We can rename tests yaml file to achieve it.

**Pull request closed without merging**